services/slam/slam.go

```diff
@@ -82,6 +84,40 @@ type Service interface {
 	resource.Generic
 }
 
+// Helper function to grab all chunks from a streamed grpc endpoint.
+func helperConcatenateChunksToFull(f func() ([]byte, error)) ([]byte, error) {
+	var fullBytes []byte
+	for {
+		chunk, err := f()
+		if errors.Is(err, io.EOF) {
+			return fullBytes, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		fullBytes = append(fullBytes, chunk...)
+	}
+}
+
+// GetPointCloudMapFull concatenates the streaming responses from GetPointCloudMapStream into a full point cloud.
```

Comment:
do we want this to return a pointcloud or a PCD? name of function should reflect this

Reply:
I think pcd, so that the caller can parse it as a PointCloud or Octree, or leave it as a pcd. I don't have a strong opinion about the name.

Reply:
kk leaving as is